### ISSUE TITLE
📎 Handle and recover from unknown dimensions

### ISSUE THREAD

**Issue by faultyserver** [biomejs/biome#1370]
### Description

Tracing this error in the benchmarks: https://github.com/biomejs/biome/actions/runs/7355456817/job/20024092373#step:7:306

The code that caused it was:
```css

@media screen and (min-width: 0\0 ) {
  /* ... */
}
```

Specifically, the `0\0` here is not something that we support, but it is actually valid when parsing because `<dimension>` is defined as "a number followed by a unit identifier, which is an identifier" [Official Railroad Diagram](https://drafts.csswg.org/css-syntax/#typedef-dimension-token). Even though this doesn't result in a valid unit, it is still parseable and can be treated as an `UnknownDimension`, which is [what SWC does, too](https://swc-css.netlify.app/?code=eJzjcshNTclMVChOLkpNzVNIzEtR0MjNzNMtz0wpybBSMIgxUNBUqOaqBQAOIAyY).

The W3C CSS Validator calls this out as a CSS hack, but doesn't really explain more, and I haven't found anything explaining it yet (EDIT: [Here's a description of the hack](http://browserbu.gs/css-hacks/media-min-width-0-backslash-0/), but given how other tools are handling this, and the fact that the grammar technically allows it, I think we should start supporting `UnknownDimension` as well. In the future, the analyzer can then call this out and lint for it as well.

The other issue here is that the parser failed to recover in this situation and panicked that it was no longer progressing at the `)` after the unit. While not necessarily related, both issues should be fixed.
